### PR TITLE
Customise bot

### LINKED ISSUE #896: Update and enhance how-to-guide section
This issue is to track :
* What has been done and 
* What we are on currently working :)

## Team members
Abhishek Jaiswal - @abhishek-iiit 
Vasvi Sood  - @vasvi-sood 

# References 
- [x] **Readme(to github section)** 
- [x] **Overview** 
- [x] **Add events to the bot**
    - [x] Add events and customise a bot
- [x] **Add more functionality to your bot**
    - [x] Dealing with messages
    - [x] Manage Contacts
    - [x] Manage Rooms
    - [x] Send and Accept Friend Requests
    - [x] Sending and Receiving files
- [x] **Deploy on IM platfrom**
    - [x] Deploy in Wechat
    - [x] Deploy in Whatsapp
    - [x] Deploy in Lark
    - [x] Deploy in Gitter
    - [x] Deploy in Wechat Official Account
    - [x] Deploy in PadLocal
    - [x] Deploy in WeChat4U
- [x] **Overview with containers**
    - [x] Deploy with Heroku
    - [x] Deploy with Docker

Please link pull request to this section for the easy access.

### PR DESCRIPTION
**Pull request by abhishek-iiit**
customise-bot1
customise-bot2
customise-bot3
customise-bot4
customise-bot6
customise-bot7
customise-bot8
customise-bot9
customise-bot10
customise-bot11

### CODE REVIEW

File: docusaurus/docs/howto/event.md [hunk 1]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
-The Wechaty Events are all the same acrossing Polyglot Wechaty programs!
+The wechaty bot is available in various programming languages!
 
 :::
 
+Events are functions that define the operations the bot can perform. Some common events are login, logout, onMessage etc.
+
+After getting familiar with the ding dong bot, this tutorial will focus on creating a bot from scratch adding various events one by one.
+
+For simplicity, this tutorial will guide you to create a bot in javascript but but you can choose between various programming languages available.
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: Before starting this guide, make sure you are already familiar with the ding dong bot. If not, go to __. In this section, you will learn how to add various events to a bot. Events are functions that define the operations a bot can perform. Some common events include `login`, `logout`, `onMessage`. The guide will demonstrate in JavaScript, but you can choose between various programming languages available.

- Put the prerequisite at the very beginning, so if the users are not equipped with the knowledge, they could leave first.
- And why are you separate the three sentences to three paragraphs?

File: docusaurus/docs/howto/event.md [hunk 2]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
+After getting familiar with the ding dong bot, this tutorial will focus on creating a bot from scratch adding various events one by one.
+
+For simplicity, this tutorial will guide you to create a bot in javascript but but you can choose between various programming languages available.  
+
+## Getting started
+
+<ol>
+<li>Create an empty project folder</li>
+<li> Add a bot.js file</li>
+<li>In the termial write the following command</li>
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: In the terminal, type the following command:

File: docusaurus/docs/howto/event.md [hunk 3]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
+
+```bash
+npm init
+```
+
+A package.json file is created.
+
+<li>In the terminal install the necesary dependencies</li>
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: In the terminal[,] install the necesary dependencies

File: docusaurus/docs/howto/event.md [hunk 4]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
+
+<li> In the terminal type the following Commands</li>
+
+```bash
+ export WECHATY_LOG=verbose
+ ```
+
+<ul>
+
+<li> For whatsapp:</li>
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: please use their official name: WhatsApp

File: docusaurus/docs/howto/event.md [hunk 5]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
+
+```bash
+export WECHATY_PUPPET=wechaty-puppet-whatsaap
+```
+
+<li>For wechat</li>
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: WeChat

File: docusaurus/docs/howto/event.md [hunk 6]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
+
+</ul>
+
+<li>Now we can run the bot using the command:</li>
+
+``` bash
+npm start
+```
+
+But before that we need to add events to the bot. For following along the tutorial use the javascript code from the polyglot.
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: Remove this sentence and add a step in between: Add events to the bot. See the following examples.

File: docusaurus/docs/howto/event.md [hunk 7]
```diff
@@ -8,15 +8,95 @@ import TabItem from '@theme/TabItem'
...
+</ol>
+
 ## Basic Events
 
-### `scan` Event: QR code
+### `Scan` Event
```
Comment (proudofsimin) [inline docusaurus/docs/howto/event.md]: `scan` is an event. lower case, please.

**Review by proudofsimin — CHANGES_REQUESTED** [review]
1. Please correct all the typos and add colons/ commas you forgot (I didn't point out all of them).
2. Please remove the *TODO* in each event and replace them with short introductions of the events' functions. Like what you did in the `scan` event is great. (but don't forget the `` tag and a period at the end.)

File: docusaurus/docs/howto/event.md (hunk 1 above)
Comment (abhishek-iiit) [reply docusaurus/docs/howto/event.md]: Thanks a lot, Simin for the suggested changes, we will get improvement done.
We broke into paragraphs so as to keep it as short as possible (not to look like a lengthy one).

Comment (proudofsimin) [reply docusaurus/docs/howto/event.md]: I'm sorry my bad. I forgot it. Could you please point out where to review next time for me ? Many thanks. Then just correct the typos and this PR is done.

Comment (abhishek-iiit) [reply docusaurus/docs/howto/event.md]: Thank you, Simin! I Will surely try to comment on the changes I did from the next PR :)

Comment (proudofsimin) [reply docusaurus/docs/howto/event.md]: > Before starting this guide, make sure you are already familiar with the ding dong bot. If not, go to __. In this section, you will learn how to add various events to a bot. Events are functions that define the operations a bot can perform. Some common events include `login`, `logout`, `onMessage`. The guide will demonstrate in JavaScript, but you can choose between various programming languages available.

Ah I'm so sad. I even rewrote it for you, but why didn't you change it ?

Comment (abhishek-iiit) [reply docusaurus/docs/howto/event.md]: So sorry, I misinterpreted the message, so as to change the typo only. Will change it tonight as soon as possible .

**Review by proudofsimin — APPROVED** [review]
Thank you :)

**Comment by abhishek-iiit** [comment]
@Rohitesh-Kumar-Jain I request you to please review this PR :)

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
LGTM

**Comment by Rohitesh-Kumar-Jain** [comment]
@vasvi-sood  We would be glad if you may once go through the PR, and either request for changes or approve the PR.

team-mates need to approve each other's PRs to get them merged.

File: docusaurus/docs/howto/event.md [hunk 8]
```diff
@@ -8,15 +8,91 @@ import TabItem from '@theme/TabItem'
...
+npm start
+```
+
+But, before that we need to add events to the bot. Following are the below Events:
+</ol>
+
 ## Basic Events
 
-### `scan` Event: QR code
+### `scan` Event
```
Comment (vasvi-sood) [inline docusaurus/docs/howto/event.md:93]: Add the description of the events.

**Review by vasvi-sood — CHANGES_REQUESTED** [review]
Make the changes before it can be merged.

**Comment by vasvi-sood** [comment]
@Rohitesh-Kumar-Jain and @proudofsimin all changes have been made. Kindly review it and add the ready to merge label.

**Review by vasvi-sood — APPROVED** [review]

**Comment by abhishek-iiit** [comment]
Thankyou Vasvi for putting up the changes :)

**Review by huan — APPROVED** [review]
LGTM.

Please read and follow #1257, and feel free to help yourself to merge it by following the merge workflow.

Thank you very much!

**Comment by Rohitesh-Kumar-Jain** [comment]
@abhishek-iiit resolve conflicts and then merge

**Comment by abhishek-iiit** [comment]
> @abhishek-iiit resolve conflicts and then merge

@Rohitesh-Kumar-Jain resolved conflict issue :)

**Comment by abhishek-iiit** [comment]
> @abhishek-iiit resolve conflicts and then merge

Conflict resolved and merging now :)

**Pull request merged**